prow/flagutil/kubernetes_cluster_clients.go

```diff
@@ -174,16 +175,33 @@ func (o *KubernetesOptions) BuildClusterClients(namespace string, dryRun bool) (
 		return nil, err
 	}
 
+	buildClusterAliases := sets.NewString()
 	buildClients := map[string]corev1.PodInterface{}
-	for context, client := range o.kubernetesClientsByContext {
-		// we want to map build cluster clients by their alias, not their context
-		// the only context that is not the alias is the default context, so
-		// we can simply overwrite that one and be content that our build cluster
-		// mapping is correct for the controller
-		if context == o.infraContext {
-			context = kube.DefaultClusterAlias
-		}
-		buildClients[context] = client.CoreV1().Pods(namespace)
+	for alias, client := range contextsToAliases(o.kubernetesClientsByContext, o.infraContext) {
+		buildClusterAliases.Insert(alias)
+		buildClients[alias] = client.CoreV1().Pods(namespace)
+	}
+	if !buildClusterAliases.HasAll(knownAliases.UnsortedList()...) {
+		return nil, fmt.Errorf("the following cluster aliases declared for jobs were not found in loaded clusters: %v", knownAliases.Difference(buildClusterAliases).List())
```

Comment:
Almost makes me think we should do this on config-load as well ... or mount the cluster credentials into the `checkconfig` job and do it pre-merge as well. @cjwagner WDYT?